src/Validation/ResponseValidator.php

```diff
@@ -188,9 +188,18 @@ protected function wrapAttributesToArray($properties)
                 unset($attributes->nullable);
             }
 
-            if ($attributes->type === 'object') {
+            if ($attributes->type === 'object' && isset($attributes->properties)) {
```

Comment:
I ran into some issue testing this on a real test suite, there is a case where `properties` are not defined in a specification file.

Reply:
Do we need more test variations?

Reply:
I checked again where this issue occurred. It's when a response is added for a path and it's defined like this:

````
"422": {
    "description": "Unprocessable Entity",
    "content": {
        "application/json": {
            "schema": {
                "type": "object",
                "properties": {
                    "message": {
                        "type": "string",
                        "description": "The error message that occurred",
                        "example": "The given data was invalid."
                    },
                    "errors": {
                        "description": "The errors that occurred",
                        "type": "object", <--- This has no "properties" key
                        "example": {
                            "username": ["This username has already been taken."]
                        }
                    }
                }
            }
        }
    }
}
````
In this example it's in the Laravel validation format and is reused for other paths too as `#/components/responses/422`. I'm not sure if there is supposed to be a "properties" key there because it's different for every request so you can't really define a schema that works for all of them.

If you want I'll add this as a test variation so that extra check is covered too.

Reply:
@hotmeteor Do you think there should be a test for the example above?

Reply:
Sorry, never followed up. I guess I'm a little concerned about supporting the depth of this, seems like it can just become recursive. 

Additionally, `nullable` is obsolete starting in OAS 3.1, it's now represented as something like:
```json
"type": ["object", "null"],
```

I'm actually wondering about what was fixed before, because there's definitely tests for nullable values in both 3.0 and 3.1. See https://github.com/hotmeteor/spectator/blob/master/tests/ResponseValidatorTest.php#L119

Unless I'm misunderstanding?

Reply:
This PR would indeed support recursive updating of nullable properties when the `type` of a property is `array` and its items are `object` again.

However the extra test here would be for ` isset($attributes->properties)` that prevents trying to loop over parts of responses or schema's that don't have a "properties" property defined like in my example. In the example it's a response that is reused as a component so there can't be any "properties" defined that works everywhere the component is used. But without the extra check the validator is trying to recursively update any `$attributes->properties` that don't exist and throws an error on it beause it's not set on `$attributes`.

I hope that makes more sense.

Reply:
OK that's more clear. Thanks.